go/pserver/service.go

```diff
@@ -49,13 +53,58 @@ type Service struct {
 	optMap map[string]*optimizer
 }
 
+// Checkpoint saves the checkpoint for pserver
+type Checkpoint struct {
+	UUID      string `json:"uuid"`
+	MD5       string `json:"md5"`
+	Timestamp string `json:"timestamp"`
+	State     []byte
+	ParameterWithConfig
```

Comment:
Rename `ParameterCheckpoint` to `CheckpointData` should be more clear. Then these two struct should be:

```go
type CheckpointMeta struct {}
type CheckpointData struct {}
type Checkpoint struct {
  meta CheckpointMeta
  data CheckpointData
}
```